src/ghci/mod.rs

```diff
@@ -127,13 +127,15 @@ impl GhciOpts {
     pub fn from_cli(opts: &Opts) -> miette::Result<(Self, Option<DuplexStream>)> {
         // TODO: implement fancier default command
         // See: https://github.com/ndmitchell/ghcid/blob/e2852979aa644c8fed92d46ab529d2c6c1c62b59/src/Ghcid.hs#L142-L171
-        let command = opts
-            .command
-            .clone()
-            .unwrap_or_else(|| ClonableCommand::new("cabal").arg("repl"));
+        let command = match (&opts.file, &opts.command) {
+            (Some(file), None) => ClonableCommand::new("ghci").arg(file.relative()),
+            (None, Some(command)) => command.clone(),
+            (None, None) => ClonableCommand::new("cabal").arg("repl"),
+            (Some(_), Some(_)) => unreachable!(),
+        };
```

Comment:
Let me know if there's a nicer way to write this.

Reply:
Maybe like `opts.command.clone().or(opts.file.map(|file| ClonableCommand::new("ghci").arg(file.relative()))).unwrap_or_else(|| ClonableCommand::new("cabal").arg("repl"))`?

Reply:
(That will probably look better once it's formatted.)

Reply:
Hmm, IMO that's less readable. I'll just keep it as is.

![D75ZK0Y8](https://github.com/MercuryTechnologies/ghciwatch/assets/887196/1e69b89a-18ff-4caf-a823-bdd4bf085516)